src/nvim/charset.c

```diff
@@ -545,18 +545,8 @@ void transchar_nonprint(char_u *buf, int c)
     buf[1] = (char_u)(c ^ 0x40);
 
     buf[2] = NUL;
-  } else if (c >= 0x80) {
```

Comment:
Based on the Vim code it appears that `|x` and `~x` were intended for non-multibyte builds or when `&encoding` is not UTF-8: not actually a dead code, just we removed all conditions which may trigger it.